firedrake/trefftz.py

```diff
@@ -0,0 +1,231 @@
+"""
+This module provides a class to compute the Trefftz embedding of a given function space.
+It is also used to compute aggregation embedding of a given function space.
+"""
+from firedrake.petsc import PETSc
+from firedrake.cython.dmcommon import FACE_SETS_LABEL, CELL_SETS_LABEL
+from firedrake.assemble import assemble
+from firedrake.mesh import Mesh
+from firedrake.functionspace import FunctionSpace
+from firedrake.function import Function
+from firedrake.ufl_expr import TestFunction, TrialFunction
+from firedrake.constant import Constant
+from ufl import dx, dS, inner, jump, grad, dot, CellDiameter, FacetNormal
+import scipy.sparse as sp
+
+
```

Comment:
Please define `__all__`  to avoid pollution of the namespace

Reply:
Or equivalently you can replace `from firedrake.trefftz import *` with `from firedrake.trefftz import OnlyWhatIWant` inside `__init__.py`.